codespan-reporting/src/lib.rs

```diff
@@ -84,3 +85,67 @@ impl Severity {
         }
     }
 }
+
+/// A command line argument that configures the coloring of the output
+///
+/// This can be used with command line argument parsers like `clap` or `structopt`.
+///
+/// # Example
+///
+/// ```rust
+/// extern crate codespan_reporting;
+/// #[macro_use]
+/// extern crate structopt;
+///
+/// use structopt::StructOpt;
+/// use codespan_reporting::termcolor::StandardStream;
+/// use codespan_reporting::ColorArg;
+///
+/// #[derive(Debug, StructOpt)]
+/// #[structopt(name = "pikelet")]
+/// pub struct Opts {
+///     /// Configure coloring of output
+///     #[structopt(
+///         long = "color",
+///         parse(try_from_str),
+///         default_value = "auto",
+///         raw(possible_values = "ColorArg::VARIANTS", case_insensitive = "true")
+///     )]
+///     pub color: ColorArg,
+/// }
+///
+/// fn main() {
+///     let opts = Opts::from_args();
+///     let writer = StandardStream::stderr(opts.color.into());
+/// }
+/// ```
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub struct ColorArg(pub ColorChoice);
+
+impl ColorArg {
+    /// Allowed values the argument
+    ///
+    /// This is useful for generating documentation via `clap` or `structopt`'s
+    /// `possible_values` configuration.
+    pub const VARIANTS: &'static [&'static str] = &["auto", "always", "ansi", "never"];
+}
+
+impl FromStr for ColorArg {
+    type Err = &'static str;
+
+    fn from_str(src: &str) -> Result<ColorArg, &'static str> {
+        match src {
+            _ if src.eq_ignore_ascii_case("auto") => Ok(ColorArg(ColorChoice::Auto)),
+            _ if src.eq_ignore_ascii_case("always") => Ok(ColorArg(ColorChoice::Always)),
+            _ if src.eq_ignore_ascii_case("ansi") => Ok(ColorArg(ColorChoice::AlwaysAnsi)),
```

Comment:
Do you think this should be `always-ansi`?

Reply:
I think it is fine to just have it as `ansi`, `AlwaysAnsi` in the enum seems rather redundant in the firstplace but perhaps it is better to be consistent and use `always-ansi`. Not sure.

Reply:
FWIW `rg` has just `ansi` on the command line.

Reply:
Ah, might just go with `rg` then - I did notice you had `always-ansi` in Gluon. I wonder how much time has been wasted on this same question around the world! 😆